src/main/java/group1/mips_simulator/components/Computer.java

```diff
@@ -19,12 +20,11 @@ public class Computer {
 
     public Computer() {
         cpu = new CPU();
-        mem = new Memory();
+        mem = new Memory(ComputerConfig.memorySize);
     }
 
     public void executeProgram(Vector<Instruction> program) {
-        // IF -> ID -> Exe ->
-        //
+        // IF -> ID -> EX -> ME -> WB
```

Comment:
I'm not quite sure what this means. Can you add more human readable documentation here?

Reply:
oh mb, did not mean to commit this, this was just a personal note for planning steps

(for future ref tho:
```
IF = Instruction fetch
ID = Instruction decode
EX = Execute Instruction
ME = Access Memory
WB = Write Back (to registers)
```
its shorthand for the pipeline stages in a MIPS architecture
The idea here is these can all theoretically be converted into Units, i.e. Instruction Fetch Unit, that do each of these specialized tasks; that way we can constrain the number of inputs/outputs each of them need/create to exactly what the hardware would need/create, and our executeProgram would (along with the try/except ofc) look something long the lines of:
```
public void executeProgram(Vector<Instruction> program) {
    for(Instruction instruction : program) {    // Fetch Instruction
        Decode Instruction
        Execute Instruction
        Access Memory
        Write to Registers
```
which would have been useful if we needed to implement pipelining, but I have since realized that we dont
still, the flow of events could be useful in informing how we execute our instructions, as we start implementing them

Reply:
Sweet! Can you add some of this to the file as in line comments?